agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -946,10 +875,34 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
             bss_info_conf_list.push_back(bss_info_conf);
         }
 
-        ap_wlan_hal->update_vap_credentials(bss_info_conf_list);
+        ap_wlan_hal->update_vap_credentials(bss_info_conf_list, backhaul_vaps_list);
 
         break;
     }
+    case beerocks_message::ACTION_APMANAGER_START_WPS_PBC_REQUEST: {
+        LOG(DEBUG) << "Got ACTION_APMANAGER_START_WPS_PBC_REQUEST";
+        auto request =
+            beerocks_header->addClass<beerocks_message::cACTION_APMANAGER_START_WPS_PBC_REQUEST>();
+        if (!request) {
+            LOG(ERROR) << "addClass cACTION_APMANAGER_START_WPS_PBC_REQUEST failed";
+            return false;
+        }
+        const auto &vaps = ap_wlan_hal->get_radio_info().available_vaps;
+        auto it          = std::find_if(
+            vaps.begin(), vaps.end(), [&](const std::pair<int, bwl::VAPElement> &element) {
+                return network_utils::mac_from_string(element.second.mac) == request->bssid();
+            });
+        if (it == vaps.end()) {
+            LOG(ERROR) << "Failed to find VAP with bssid " << request->bssid();
+            return false;
```

Comment:
I don't think you should return `false` here (it's for critical errors only).
If there's no response message, just `break`.